File: Aula8/Matriz.java
import java.util.ArrayList;

public class Matriz{
    /*
    *Método que inicializa uma matriz de inteiros.
    *@param matriz -> Matriz de numeros inteiros.
    *@param qtdl -> Quantidade de linhas.
    *@param qtdc -> Quantidade de colunas.
    */
    public static void incializarMatriz(int matriz[][], int qtdl, int qtdc){
        for(int i = 0; i< qtdl; i++){
            for(int j = 0; j<qtdc; j++){
                matriz[i][j] = 0;
            }
        }
    }

    /*
    *Método que exibe uma matriz de inteiros.
    *@param matriz -> Matriz de de numeros inteiros.
    *@param qtdl -> Quantidade de linhas.
    *@param qtdc -> Quantidade de colunas.
    */
    public static void exibirMatriz(int matriz[][], int qtdl, int qtdc){
        for(int i = 0; i < qtdl; i++){
            for(int j = 0; j < qtdc; j++){
                System.out.print(matriz[i][j] + "\t");
            }
            System.out.println();
        }
    }


    /*
    *Método que converte uma matriz de inteiros para uma lista.
    *@param matriz -> Matriz de numeros inteiros.
    *@param qtdl -> Quantidade de linhas.
    *@param qtdc -> Quantidade de colunas.
    *@param lista -> Lista de inteiros que recebe os valores convertidos.
    */
    public static void converterMatrizLista(int matriz[][], int qtdl, int qtdc, ArrayList<Dado> lista){
        for(int i = 0; i< qtdl; i++){
            for(int j = 0; j < qtdc; j++){
                if(matriz[i][j] != 0){
                    lista.add(new Dado(matriz[i][j], i, j));
                }
            }
        }
    }


    /*Método que exibe uma lista de objetos Dado.
    *@param lista -> Lista de objetos do tipo Dado.
    *
    */
    public static void exibirLista(ArrayList<Dado> lista){
        for(Dado dado : lista){
            System.out.println(dado);
        }
        System.out.println("-------------------------------");
        System.out.println("Total de elementos nao nulos " + lista.size());
    }


}


    

File: Aula8/PYTHON/Matriz.py
from Dado import Dado
class Matriz:

    """Função que inicializa uma matriz com zeros.
        
    Args:
        matriz (list): A matriz a ser inicializada
        linhas (int): Número de linhas da matriz
        colunas (int): Número de colunas da matriz
    """
    def inicializarMatriz(matriz, linhas, colunas):
        for i in range(linhas):
            for j in range(colunas):
                matriz[i][j] = 0

    """Função que exibe os elementos de uma matriz.
        
    Args:
        matriz (list): A matriz a ser exibida
        linhas (int): Número de linhas da matriz
        colunas (int): Número de colunas da matriz
    """
    def exibirMatriz(matriz, linhas, colunas):
        for i in range(linhas):
            for j in range(colunas):
                print(matriz[i][j], end='  ')
            print()
    
    """Função que converte os elementos de uma matriz em uma lista.
        
    Args:
        matriz (list): A matriz a ser convertida
        linhas (int): Número de linhas da matriz
        colunas (int): Número de colunas da matriz
        lista (list): A lista a ser preenchida
    """
    def converterMatrizLista(matriz, linhas, colunas, lista):
        for i in range(linhas):
            for j in range(colunas):
                if matriz[i][j] != 0:
                    lista.append(Dado(matriz[i][j], i, j))    

    def exibirLista(lista):
        for item in lista:
            print(item, end=' ')
        print()
        print(f"Total de elementos não nulos: {len(lista)}")